### PR TITLE
[MM-24435] refactor terraform.Output to be more dev friendly

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This PR refactors the `terraform.Output` and removes `Value` calls. This is mostly cosmetic changes but it was bothering long ago.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24435

### CODE REVIEW

File: deployment/terraform/output.go [hunk 1]
```diff
@@ -3,91 +3,149 @@
...
+
+type Instance struct {
+	PrivateIP  string `json:"private_ip"`
+	PublicIP   string `json:"public_ip"`
+	PublicDNS  string `json:"public_dns"`
+	PrivateDNS string `json:"private_dns"`
+	Tags       Tags   `json:"tags"`
+}
+
+type ElasticsearchDomain struct {
```
Comment (agnivade) [inline deployment/terraform/output.go:59]: ElasticSearchDomain

File: deployment/terraform/output.go [hunk 2]
```diff
@@ -3,91 +3,149 @@
...
 }
 
 //HasElasticSearch returns whether a deployment has ElasticSaearch installed in it or not.
 func (o *Output) HasElasticSearch() bool {
-	return len(o.ElasticServer.Value) > 0
+	return o.ElasticSearchServer.Endpoint != ""
 }
 
-// HasMetrics returns whether a deployment includes app server instances.
+// HasAppServers returns whether a deployment includes app server instances.
```
Comment (streamer45) [inline deployment/terraform/output.go:133]: Thanks :+1:

File: deployment/terraform/output.go [hunk 3]
```diff
@@ -3,91 +3,149 @@
...
+	return len(o.Instances) > 0
 }
 
 // HasMetrics returns whether a deployment includes the metrics instance.
 func (o *Output) HasMetrics() bool {
-	return len(o.MetricsServer.Value) > 0
+	return o.MetricsServer.PrivateIP != ""
+}
+
+// HasS3Bucket returns whether a deployment includes the the S3 Bucket.
```
Comment (streamer45) [inline deployment/terraform/output.go:143]:
```suggestion
// HasS3Bucket returns whether a deployment includes the S3 Bucket.
```

File: deployment/terraform/output.go [hunk 4]
```diff
@@ -3,91 +3,149 @@
...
+}
+
+// Output reads the current terraform output
+func (t *Terraform) Output() (*Output, error) {
+	var buf bytes.Buffer
+	err := t.runCommand(&buf, "output", "-json")
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (streamer45) [inline deployment/terraform/output.go:90]:
```suggestion
	if err := t.runCommand(&buf, "output", "-json"); err != nil {
		return nil, err
	}
```

:police_officer: :stuck_out_tongue:

File: deployment/terraform/output.go [hunk 5]
```diff
@@ -3,91 +3,149 @@
...
+	var o output
+	err = json.Unmarshal(buf.Bytes(), &o)
+	if err != nil {
+		return nil, err
+	}
```
Comment (streamer45) [inline deployment/terraform/output.go:95]:
```suggestion
	if err := json.Unmarshal(buf.Bytes(), &o); err != nil {
		return nil, err
	}
```

File: deployment/terraform/output.go (hunk 4 above)
Comment (agnivade) [reply deployment/terraform/output.go]: Can we just call this the Claudio coding-style? :laughing:

**Pull request merged**